lib/logic.rb

```diff
@@ -0,0 +1,149 @@
+class Board
+  attr_accessor :board
+
+  def initialize
+    @turn = true
+    @board = [
+      [' ', ' ', ' '],
+      [' ', ' ', ' '],
+      [' ', ' ', ' ']
+    ]
+  end
+
+  def print_board
+    @board.each do |row|
+      puts row.to_s
+    end
+  end
+
+  def full?
+    @board.each do |row|
```

Comment:
Can we not use a nested `each` here? If our board is simply an array we don't need to put an `each` inside another `each`. We can simply iterate through the array and check for what you need to check.

For instance, if you follow my suggestion to use something like [1,2,3,4,5,6,7,8,9] for your @board variable then you can simply iterate over each element of @board to check whether an element is already either "X" or "O".